feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.spec.ts

```diff
@@ -55,6 +71,57 @@ const mockActivatedRoute = {
   },
 };
 
+describe('CheckoutOnePaymentTypeComponent', () => {
```

Comment:
I created separate CheckoutPaymentTypeComponent because I have to use different class for provide: CheckoutPaymentTypeFacade if there is way to use difference class for same provider then I can change it.

Reply:
Yes, it's possible.  For a given test
- use subjects that are manually controlled
- you can override the mock's return value using a `jasmine.Spy`
- override the TestBed configuration with `TestBed.overrideProvider()` https://angular.io/api/core/testing/TestBed#overrideprovider

Unfortunately, for this component, to do either requires a restructuring of the spec file.  I would recommend restructuring the spec file, but we should wait for word from a Spartacus team member first.

Reply:
As there was no followup from the Spartacus reviewers, I will consider this accepted and resolved.